### PR TITLE
Add web test client restdocs autoconfiguration

### LINKED ISSUE #10967: Support WebTestClient in AutoConfigureRestDocs
At the moment `@AutoConfigureRestDocs` only supports Mock MVC and REST Assured. It would be nice to have WebTestClient (webflux) support there too.

Spring boot version: v2.0.6.M6

### PR DESCRIPTION
**Pull request by zaynetro**
Closes #10967

This is my work in progress to autoconfigure WebTestClient.

### CODE REVIEW

File: spring-boot-project/spring-boot-test-autoconfigure/src/main/java/org/springframework/boot/test/autoconfigure/restdocs/RestDocsAutoConfiguration.java [hunk 1]
```diff
@@ -108,4 +110,26 @@ public RestDocsRestAssuredBuilderCustomizer restAssuredBuilderCustomizer(
 
 	}
 
+	@Configuration
+	@ConditionalOnClass(WebTestClientRestDocumentation.class)
+	@ConditionalOnWebApplication(type = Type.SERVLET)
```
Comment (eddumelendez) [inline spring-boot-project/spring-boot-test-autoconfigure/src/main/java/org/springframework/boot/test/autoconfigure/restdocs/RestDocsAutoConfiguration.java]: it should be reactive

Comment (zaynetro) [reply spring-boot-project/spring-boot-test-autoconfigure/src/main/java/org/springframework/boot/test/autoconfigure/restdocs/RestDocsAutoConfiguration.java]: Thanks!

**Comment by zaynetro** [comment]
How do I execute one test only?

Specifically `org.springframework.boot.test.autoconfigure.restdocs.WebTestClientRestDocsAutoConfigurationIntegrationTests`

**Comment by wilkinsona** [comment]
@zaynetro You can either run that test in your IDE, or use `-Dtestcase` on the command line

**Comment by zaynetro** [comment]
I figured that with this change test succeeds:

```diff
diff --git a/spring-boot-project/spring-boot-test-autoconfigure/src/main/java/org/springframework/boot/test/autoconfigure/web/reactive/WebTestClientAutoConfiguration.java b/spring-boot-project/spring-boot-test-autoconfigure/src/main/java/org/springframework/boot/test/autoconfigure/web/reactive/WebTestClientAutoConfiguration.java
index 8997ebb4bd..68264a8e3d 100644
--- a/spring-boot-project/spring-boot-test-autoconfigure/src/main/java/org/springframework/boot/test/autoconfigure/web/reactive/WebTestClientAutoConfiguration.java
+++ b/spring-boot-project/spring-boot-test-autoconfigure/src/main/java/org/springframework/boot/test/autoconfigure/web/reactive/WebTestClientAutoConfiguration.java
@@ -30,6 +30,7 @@ import org.springframework.context.ApplicationContext;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 import org.springframework.http.codec.ClientCodecConfigurer;
+import org.springframework.restdocs.webtestclient.WebTestClientRestDocumentationConfigurer;
 import org.springframework.test.web.reactive.server.WebTestClient;
 import org.springframework.test.web.reactive.server.WebTestClient.Builder;
 import org.springframework.util.CollectionUtils;
@@ -49,11 +50,13 @@ public class WebTestClientAutoConfiguration {

        @Bean
        @ConditionalOnMissingBean
-       public WebTestClient webTestClient(ApplicationContext applicationContext) {
+       public WebTestClient webTestClient(ApplicationContext applicationContext, WebTestClientRestDocumentationConfigurer configurer) {
                WebTestClient.Builder builder = WebTestClient
                                .bindToApplicationContext(applicationContext).configureClient();
                customizeWebTestClient(builder, applicationContext);
                customizeWebTestClientCodecs(builder, applicationContext);
+               builder.baseUrl("https://api.example.com/");
+               builder.filter(configurer);
                return builder.build();
        }
```

I know it isn't not the proper way to set up the WebTestClient. Due to the lack of knowledge I don't know how to add Configurer only if RestDocs are setup. 

Could someone point me at how I can overcome this and whether I am doing it at the right place?

File: spring-boot-project/spring-boot-parent/src/checkstyle/checkstyle.xml [hunk 2]
```diff
@@ -70,7 +70,7 @@
...
....springframework.boot.test.autoconfigure.AutoConfigurationImportedCondition.*, org.mockito.Mockito.*, org.mockito.BDDMockito.*, org.mockito.ArgumentMatchers.*, org.mockito.Matchers.*, org.springframework.restdocs.hypermedia.HypermediaDocumentation.*, org.springframework.restdocs.mockmvc.MockMvcRestDocumentation.*, org.springframework.restdocs.webtestclient.WebTestClientRestDocumentation.*, org.springframework.restdocs.operation.preprocess.Preprocessors.*, org.springframework.restdocs.restassured3.operation.preprocess.RestAssuredPreprocessors.*, org.springframework.restdocs.restassured3.RestAssuredRestDocumentation.*, org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*, org.springframework.test.web.servlet.result.MockMvcResultMatchers.*, org.springframework.security.test.web.servlet.request.SecurityMockMvcRequestPostProcessors.*, org.springframework.security.test.web.servlet.setup.SecurityMockMvcConfigurers.*, org.springframework.hateoas.mvc.ControllerLinkBuilder.linkTo, org.springframework.test.web.client.ExpectedCount.*, org.springframework.test.web.client.match.MockRestRequestMatchers.*, org.springframework.test.web.client.response.MockRestResponseCreators.*" />
```
Comment (zaynetro) [inline spring-boot-project/spring-boot-parent/src/checkstyle/checkstyle.xml]: Added `org.springframework.restdocs.webtestclient.WebTestClientRestDocumentation.*`

**Comment by zaynetro** [comment]
Figured it out.

**Comment by zaynetro** [comment]
Yay! Let me know if something else needs to be changed.

**Comment by zaynetro** [comment]
Any chance this can get merged some time soon?

**Comment by wilkinsona** [comment]
@zaynetro Thanks for the PR. I'll try to review and merge this in RC1.

**Comment by wilkinsona** [comment]
Thanks for the PR, @zaynetro. I've merged the changes into master.

**Pull request closed without merging**